### PR TITLE
feat: upgrade MiniMax default model to M3 (512K context, 128K output)

### PR DESCRIPTION
**Pull request by octo-patch**
## Summary

This PR adds first-class support for the [MiniMax-M3](https://huggingface.co/MiniMaxAI/MiniMax-M3) architecture in OpenMythos:

- **`minimax_m3_config()`** — a new `MythosConfig` factory in `open_mythos/variants.py` whose structural dimensions mirror MiniMax-M3's design: `dim=6144`, 48 query / 8 KV heads (6× GQA), `head_dim=128` via MLA rope+nope split (64+64), 32 routed + 2 shared experts with top-4 activation, **524 288-token (512K) context**, **131 072-token (128K) max output**, and `rope_theta=10_000_000` for long-context stability.
- **`MINIMAX_M3_MODEL_ID`** — a constant (`"MiniMaxAI/MiniMax-M3"`) added to `open_mythos/tokenizer.py` for convenient use with `MythosTokenizer`, whose `vocab_size=200064` matches the config.
- Both symbols are re-exported from the top-level `open_mythos` package.
- **`tests/test_minimax_m3.py`** — unit tests covering config dimensions (including the new 512K context window and 128K max output), MoE structure (32 routed + 2 shared + top-4), MLA cache compression, LTI spectral radius stability, high `rope_theta` numerical safety, and full forward-pass / generation correctness with synthetic tensors on CPU.

All new tests are designed to run on CPU with synthetic tensors. No changes to existing code paths.

## Why MiniMax-M3

MiniMax-M3 is the latest MiniMax release, with a 512K context window, up to 128K output tokens, and image-input support. Replacing the earlier draft (M2.7) with M3 keeps OpenMythos's reference config aligned with the current default model.

## Usage

```python
from open_mythos import minimax_m3_config, OpenMythos, MINIMAX_M3_MODEL_ID
import torch

cfg = minimax_m3_config()
model = OpenMythos(cfg)
ids = torch.randint(0, cfg.vocab_size, (1, 16))
logits = model(ids, n_loops=4)  # (1, 16, 200064)

# To use the MiniMax-M3 tokenizer:
from open_mythos.tokenizer import MythosTokenizer
tok = MythosTokenizer(MINIMAX_M3_MODEL_ID)
```

### CODE REVIEW

**Comment by Qodo-Free-For-OSS** [comment]
Hi, open_mythos.__init__.__all__ includes "load_tokenizer" and "get_vocab_size", but neither symbol is imported into the package nor defined anywhere in the open_mythos package, causing AttributeError (and potentially failing `from open_mythos import *`).

**Severity:** action required | **Category:** correctness

**How to fix:** Define or remove exported names

**Agent prompt to fix** - you can give this to your LLM of choice:

> ## Issue description
> `open_mythos/__init__.py` lists `load_tokenizer` and `get_vocab_size` in `__all__`, but these symbols are not available on the package module. This breaks star-imports and any code expecting these public APIs.
> 
> ## Issue Context
> The package currently imports `MythosTokenizer` and constants from `open_mythos.tokenizer`, but does not provide `load_tokenizer`/`get_vocab_size` wrappers.
> 
> ## Fix Focus Areas
> - open_mythos/__init__.py[18-58]
> - open_mythos/tokenizer.py[1-80]
> 
> ### Suggested implementation direction
> Either:
> 1) Remove `load_tokenizer` and `get_vocab_size` from `__all__` if they are not supported.
> 
> Or:
> 2) Implement `load_tokenizer(model_id: str = DEFAULT_MODEL_ID) -> MythosTokenizer` and `get_vocab_size(model_id: str = DEFAULT_MODEL_ID) -> int` in `open_mythos/tokenizer.py`, then import them in `open_mythos/__init__.py` and keep them in `__all__`.

---
*Found by [Qodo](https://github.com/marketplace/qodo-merge-pro-for-open-source) code review. FYI, Qodo is free for open-source.*
